src/web/app/actions/camera/preview-helper/BasePreviewManager.ts

```diff
@@ -57,6 +59,75 @@ class BasePreviewManager implements PreviewManager {
     throw new Error('Method not implemented.');
   };
 
+  // for Beam Series, BB2
+  previewRegionFromPoints = async (
+    x1: number,
+    y1: number,
+    x2: number,
+    y2: number,
+    {
+      getPoints = () => [],
+      overlapRatio = 0.05,
+    }: {
+      getPoints?: () => Array<{ point: [number, number]; overlapFlag: number }>;
+      overlapRatio?: number;
+    } = {}
+  ): Promise<boolean> => {
+    const points = getPoints();
+
+    let isStopped = false;
+    const triggerPause = ({ code }: KeyboardEvent) => {
+      if (code === 'Escape') {
+        isStopped = true;
+      }
+    };
+    const unregisterPause = shortcuts.on(['esc'], triggerPause);
+
+    try {
+      for (let i = 0; i < points.length; i += 1) {
+        if (this.ended) return false;
+
+        MessageCaller.openMessage({
+          key: 'camera-preview',
+          content: `${i18n.lang.topbar.preview} ${i}/${points.length}
+          (${i18n.lang.topbar.preview_press_esc_to_stop})`,
+          level: MessageLevel.LOADING,
+          style: { whiteSpace: 'pre-line' },
+          duration: 20,
+        });
+
+        const { point, overlapFlag } = points[i];
+        // eslint-disable-next-line no-await-in-loop, @typescript-eslint/no-explicit-any
+        const result = await (this.preview as any)(point[0], point[1], {
```

Comment:
I would recommend to change `preview` definition in `BasePreviewManager.ts` and `PreviewManager.d.ts`